src/renderer/components/ft-subscribe-button/ft-subscribe-button.js

```diff
@@ -169,30 +165,34 @@ export default defineComponent({
     },
 
     handleUnsubscription: function (profile) {
-      const currentProfile = deepCopy(profile)
-      currentProfile.subscriptions = currentProfile.subscriptions.filter((channel) => {
-        return channel.id !== this.channelId
-      })
-
-      this.updateProfile(currentProfile)
-      showToast(this.$t('Channel.Channel has been removed from your subscriptions'))
+      const profileIds = [profile._id]
 
       if (profile._id === MAIN_PROFILE_ID) {
         // Check if a subscription exists in a different profile.
         // Remove from there as well.
-        let duplicateSubscriptions = 0
 
         this.profileList.forEach((profileInList) => {
           if (profileInList._id === MAIN_PROFILE_ID) {
             return
           }
-          duplicateSubscriptions += this.unsubscribe(profileInList, this.channelId)
+
+          if (this.isProfileSubscribed(profileInList)) {
+            profileIds.push(profileInList._id)
+          }
         })
+      }
 
-        if (duplicateSubscriptions > 0) {
-          const message = this.$t('Channel.Removed subscription from {count} other channel(s)', { count: duplicateSubscriptions })
-          showToast(message)
-        }
+      this.removeChannelFromProfiles({
+        channelId: this.channelId,
+        profileIds
+      })
+
+      showToast(this.$t('Channel.Channel has been removed from your subscriptions'))
+
+      if (profile._id === MAIN_PROFILE_ID && profileIds.length > 1) {
+        showToast(this.$t('Channel.Removed subscription from {count} other channel(s)', {
```

Comment:
**suggestion (non-blocking, minor):** Can do as a separate PR, as this is a pre-existing "issue", but we might as well pluralize the label instead of doing `(s)`